crates/rpc/src/jsonrpc/websocket/data.rs

```diff
@@ -0,0 +1,215 @@
+use crate::jsonrpc::{RequestId, RpcError, RpcResponse};
+use pathfinder_common::BlockHash;
+use pathfinder_common::BlockNumber;
+use pathfinder_common::BlockTimestamp;
+use pathfinder_common::GasPrice;
+use pathfinder_common::SequencerAddress;
+use pathfinder_common::StarknetVersion;
+use pathfinder_common::StateCommitment;
+use serde::ser::Error;
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+
+use pathfinder_serde::GasPriceAsHexStr;
+use starknet_gateway_types::reply::{Block, Status};
+
+#[derive(serde::Deserialize, Serialize)]
+pub(super) struct Kind<'a> {
+    #[serde(borrow)]
+    pub(super) kind: Cow<'a, str>,
+}
+
+#[derive(Deserialize, Serialize)]
+pub(super) struct SubscriptionId {
+    pub(super) id: u32,
+}
+
+pub(super) struct SubscriptionItem<T> {
+    pub(super) subscription_id: u32,
+    pub(super) item: T,
+}
+
+impl<T: Serialize> Serialize for SubscriptionItem<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        #[derive(Serialize)]
+        struct ResultHelper<'a, U: Serialize> {
+            subscription: u32,
+            result: &'a U,
+        }
+
+        use serde::ser::SerializeMap;
```

Comment:
Huh. I would actually be interesting to know if they're equivalent? Quite a few places that would become more legible with that instead of fully specifying manually..

Reply:
This ties a serializer-agnostic piece of code to `serde_json`, while still leaving a similar amount of complexity. I'm not excited by the effort/benefit ratio.

Reply:
I'm on the fence on that particular point. There are plenty of places where we do tie the `Serialize` impl to `serde_json` and I'm guilty of this as well, but while we only expose this particular struct as JSON right now this might change (we use bincode in some other places for example). 

More importantly: I can't figure out how to make this work :smile:: `json!` gives me a `Value` that I would need to pass to the `serializer` somehow, and I don't see any way to do that without it resulting in wrapping the value in another JSON object.

Reply:
No worries! It was worth exploring.